### PR TITLE
Added support for wrapped tokens

### PR DESCRIPTION
**Pull request by mystbrent**
## Description of Changes
Added support for wrapped trust tokens.

## Environment Variables Additions
```
# TRUST TOKEN AUD
REACT_APP_TAUD_ADDRESS_MAINNET=0x00006100F7090010005F1bd7aE6122c3C2CF0090
REACT_APP_TAUD_ADDRESS_MATIC=0x5571576d39Fcb71573E0663aAA5f84B9DCDD5c90

# TRUST TOKEN CAD
REACT_APP_TCAD_ADDRESS_MAINNET=0x00000100F2A2bd000715001920eB70D229700085
REACT_APP_TCAD_ADDRESS_MATIC=0x0B76679ef27ad1473177Cb57469D7f2d03E58F47

# TRUST TOKEN GBP
REACT_APP_TGBP_ADDRESS_MAINNET=0x00000000441378008ea67f4284a57932b1c000a5
REACT_APP_TGBP_ADDRESS_MATIC=0xCE6b269081EA33B14119BB3AFd82e739BfBC37EA

# TAUD BRIDGE
REACT_APP_TAUD_MAINNET_BRIDGE_CONTRACT_ADDRESS=0x1eE3c77383aD0A0A9b8e68f7Bc6b5f5C38eCFe50
REACT_APP_TAUD_MATIC_BRIDGE_CONTRACT_ADDRESS=0xa404452E81e8602a72ECfb8E3Fc3ECC30Bf8c66f

# TCAD BRIDGE
REACT_APP_TCAD_MAINNET_BRIDGE_CONTRACT_ADDRESS=0x2Fc8B75D87Ec781AE8C02859030917473baa6968
REACT_APP_TCAD_MATIC_BRIDGE_CONTRACT_ADDRESS=0x69442f581B3D351e3898728d7f23728e6202e860

# TGBP BRIDGE
REACT_APP_TGBP_MAINNET_BRIDGE_CONTRACT_ADDRESS=0x2Fc8B75D87Ec781AE8C02859030917473baa6968
REACT_APP_TGBP_MATIC_BRIDGE_CONTRACT_ADDRESS=0x9288E93acff28a8f1A73E01D422e3d9694f5b4D3
```

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-214)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [ ] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/components/Tailwind/Modals/TokenSelectModal.tsx [hunk 1]
```diff
@@ -1,16 +1,49 @@
...
-import { HALO } from '../../../constants'
+import { HALO, MOCK, RIO, TRUE_AUD, TRUE_CAD, TRUE_GBP } from '../../../constants'
 import BaseModal from 'components/Tailwind/Modals/BaseModal'
 import CurrencyLogo from 'components/CurrencyLogo'
 import { Search } from 'react-feather'
 
 interface TokenSelectModalProps {
   isVisible: boolean
   onDismiss: () => void
+  onSelect?: (token: any | undefined) => void
```
Comment (schystz) [inline src/components/Tailwind/Modals/TokenSelectModal.tsx]: @mystbrent it looks like on this branch this param type was changed to `ChainTokenMap`, can we keep this as `Token` instead?

I was thinking the way to achieve this is by resetting the token list every time the network (`chainId`) changed - tagged you to the relevant code on my branch where supplying a custom token list would also be possible as this component is also reused on my branch (liquidity pools) & probably on @0xAplki's (swap) as well

File: src/pages/Tailwind/Bridge/BridgePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+  useEffect(() => {
+    setButtonStates()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inputValue])
+
+  useEffect(() => {
+    if (!chainId) return
+    setTokenContract(getContract(token[chainId as ChainId].address, TOKEN_ABI, library, account as string))
+    setPrimaryBridgeContract(
+      getContract(BRIDGE_CONTRACTS[token[chainId as ChainId].address], PRIMARY_BRIDGE_ABI, library, account as string)
```
Comment (schystz) [inline src/pages/Tailwind/Bridge/BridgePanel.tsx:87]: @mystbrent got a run-time error here when I select TRUE_GBP or RNBW from the token list (active network is Ethereum mainnet). Note that I don't get the error if I select TRUE_CAD

Screen Shot 2021-09-01 at 8 00 41 AM

File: src/pages/Tailwind/Bridge/index.tsx [hunk 3]
```diff
@@ -1,225 +1,10 @@
-import React, { useState } from 'react'
```
Comment (schystz) [inline src/pages/Tailwind/Bridge/index.tsx:1]: @mystbrent @tracyarciaga can we improve the UI if wallet is not connected yet? We can perhaps disable or hide some UI components - maybe a discussion with Jofo based on what's technically possible to show if wallet is not connected yet.

Screen Shot 2021-09-01 at 8 08 54 AM

Screen Shot 2021-09-01 at 8 08 59 AM

File: src/pages/Tailwind/Bridge/BridgePanel.tsx [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+      await tx.wait()
+      setApproveState(ApproveButtonState.NotApproved)
+      setButtonState(ButtonState.Default)
+      setAllowance(allowance - toNumber(amount))
+    } catch (e) {
+      console.error(e)
+      return false
+    }
+    /** Log deposit to google analytics */
+    ReactGA.event({
```
Comment (schystz) [inline src/pages/Tailwind/Bridge/BridgePanel.tsx:283]: @mystbrent oh GA already integrated, nice work!

Reminds me, need to add that too to add/remove liquidity.

**Review by schystz — COMMENTED** [review]
Some fine tuning but actual bridge functionality works fine! Great work @mystbrent !

File: src/pages/Tailwind/Bridge/index.tsx (hunk 3 above)
Comment (schystz) [reply src/pages/Tailwind/Bridge/index.tsx:1]: Spoke to Jofo, he'll just provide a design for an empty UI state if wallet is not connected yet so we can hide the UI elements (provided that is an easier/faster approach than fixing the token list & from network)

**Comment by mystbrent** [comment]
Thanks for reviewing @schystz 😄 

I'll be working the improvements in between deployment scripts and tests I'm running

File: src/pages/Tailwind/Bridge/BridgePanel.tsx (hunk 4 above)
Comment (tracyarciaga) [reply src/pages/Tailwind/Bridge/BridgePanel.tsx:283]: @schystz i added the GA :)

**Review by bitcoinbrisbane — COMMENTED** [review]
Few bits of commented out code, but otherwise LGTM

File: src/pages/Tailwind/Bridge/index.tsx (hunk 3 above)
Comment (tracyarciaga) [reply src/pages/Tailwind/Bridge/index.tsx:1]: > Spoke to Jofo, he'll just provide a design for an empty UI state if wallet is not connected yet so we can hide the UI elements (provided that is an easier/faster approach than fixing the token list & from network)

@0xAplki is assigned to to this https://halodao.atlassian.net/browse/HDEV-225

**Review by tracyarciaga — APPROVED** [review]
Finally approving this. Looks good to me now.

File: src/components/Tailwind/Modals/NetworkModal.tsx [hunk 5]
```diff
@@ -110,7 +154,14 @@ const NetworkModal = ({ isVisible, mode, onDismiss }: NetworkModalProps) => {
                   if (mode !== NetworkModalMode.SecondaryBridge) {
                     console.log('mode', mode)
                     const params = NETWORK_PARAMS[key]
-                    library?.send('wallet_addEthereumChain', [params, account])
+                    // library?.send('wallet_addEthereumChain', [params, account])
+                    if (key === ChainId.MAINNET) {
+                      library?.send('wallet_switchEthereumChain', [{ chainId: '0x1' }, account])
```
Comment (0xAplki) [inline src/components/Tailwind/Modals/NetworkModal.tsx:159]: oh this is nice!

File: src/components/Tailwind/Modals/TokenSelectModal.tsx [hunk 6]
```diff
@@ -1,16 +1,48 @@
...
+  chainId: number
+  onSelect: (token: any | undefined) => void
+}
+
+const TOKENS = [HALO, HALOHALO, TRUE_AUD, TRUE_CAD, TRUE_GBP]
+
+const TokenList = ({ chainId, onSelect }: TokenListProps) => {
+  return (
+    <div className="p-4">
+      {TOKENS.map(token => (
```
Comment (0xAplki) [inline src/components/Tailwind/Modals/TokenSelectModal.tsx:25]: we will just fix this token maps in our tickets. FYI @tracyarciaga

**Review by 0xAplki — APPROVED** [review]
left some fyis @tracyarciaga 

looks good

File: src/components/Tailwind/Modals/TokenSelectModal.tsx (hunk 1 above)
Comment (0xAplki) [reply src/components/Tailwind/Modals/TokenSelectModal.tsx]: continuing the discussion here: https://github.com/HaloDAO/halodao-interface/pull/57#pullrequestreview-744274884

File: src/components/Tailwind/Modals/TokenSelectModal.tsx (hunk 6 above)
Comment (0xAplki) [reply src/components/Tailwind/Modals/TokenSelectModal.tsx:25]: here's the conversation: https://github.com/HaloDAO/halodao-interface/pull/60#discussion_r699748777

**Pull request merged**